src/staticanalysis/bot/static_analysis_bot/task.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from cli_common.log import get_logger
+
+logger = get_logger(__name__)
+
+
+class AnalysisTask(object):
+    '''
+    An analysis CI task running on Taskcluster
+    '''
+    artifacts = []
+    valid_states = ('completed', 'failed')
+
+    def __init__(self, task_id, task_status):
+        self.id = task_id
+        assert 'task' in task_status, 'No task data for {}'.format(self.id)
+        assert 'status' in task_status, 'No status data for {}'.format(self.id)
+        self.task = task_status['task']
+        self.status = task_status['status']
+
+    @property
+    def run_id(self):
+        return self.status['runs'][-1]['runId']
+
+    @property
+    def name(self):
+        return self.task['metadata'].get('name', 'unknown')
+
+    @property
+    def state(self):
+        return self.status['state']
+
+    def load_artifacts(self, queue_service):
+
+        # Process only the supported final states
+        # as some tasks do not always have relevant output
+        if self.state not in self.valid_states:
+            logger.warn('Invalid task state', state=self.state, id=self.id, name=self.name)
+            return
+
+        # Load relevant artifacts
+        out = {}
+        for artifact_name in self.artifacts:
+            logger.info('Load artifact', task_id=self.id, artifact=artifact_name)
+            try:
+                artifact = queue_service.getArtifact(self.id, self.run_id, artifact_name)
```

Comment:
Does `getArtifact` automatically retry? We should retry a few times on failure, as they are uploaded on S3 and there could be temporary failures.

Reply:
It does up to 5 retries [when a 50x error occurs](https://github.com/taskcluster/taskcluster-client.py/blob/5ae57f12bfb56e3b7a7a150f655b8e76cb68ccbd/taskcluster/queue.py#L531)

Here is the [low level code](https://github.com/taskcluster/taskcluster-client.py/blob/43b29b63b4776360c8faf667a0182891cceaf362/taskcluster/client.py#L458) doing that in taskcluster-client.py